app/events/proxy_configs/affecting_object_changed_event.rb

```diff
@@ -12,4 +12,8 @@ def self.create(proxy, object)
       }
     )
   end
+
+  def self.valid?(proxy, *_args)
+    proxy && proxy.account&.persisted? # Proxy's service or account may have been deleted
```

Comment:
Good for me :)

It is more explicit that this event can't be valid if account is not there anymore

Though it reeks code smell by the use of `account&.` The reason is simple, do we generate config change on deletion?

Reply:
We do issue Affecting Object Changes events on deletion, yes.

Reply:
Ouch :(
Can't it be fixed when it is scheduled for deletion or maybe?
For the issue you address the PR is good enough but it looks like a can of worm if we want to fix everything.
Good for you to address (if needed) the above issue in another PR?

Reply:
@hallelujah, thinking about moving the check here to https://github.com/3scale/porta/blob/1e8bdf5d3ab494f09d2db7e471f690860f32b98f/app/lib/proxy_config_affecting_changes.rb#L13?

Reply:
https://issues.redhat.com/browse/THREESCALE-4322